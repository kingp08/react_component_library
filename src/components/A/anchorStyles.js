import styled from 'styled-components'

const StyledA = styled.a`
  color: hsl(130, 50%, 50%);
`

export default StyledA