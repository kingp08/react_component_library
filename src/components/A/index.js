import React from 'react'
import PropTypes from 'prop-types'

import StyledA from './anchorStyles'

const A = (props) => {
  return (
    <StyledA href={props.href}>
      {props.children}
    </StyledA>
  )
}

A.propTypes = {
  href: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired
}

export default A
