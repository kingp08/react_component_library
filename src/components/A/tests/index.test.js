import React from 'react'
import { shallow, mount } from 'enzyme'

import A from '..'

describe('<A />', () => {
  it('should render an <a> tag', () => {
    const wrapper = mount(<A href="#">Click me</A>)
    expect(wrapper.find('a')).toHaveLength(1)
  })

  it('should have children', () => {
    const wrapper = shallow(<A  href="#">Click me</A>)
    expect(wrapper.contains('Click me')).toEqual(true)
  })

  it('should have href attribute', () => {
    const link = 'localhost:3000'
    const wrapper = shallow(<A href={link}>Click me</A>)
    expect(wrapper).toMatchSnapshot()
  })
})