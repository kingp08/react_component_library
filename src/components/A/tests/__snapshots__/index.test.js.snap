// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<A /> should have href attribute 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <A
    href="localhost:3000"
  >
    Click me
  </A>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "Click me",
      "href": "localhost:3000",
    },
    "ref": null,
    "rendered": "Click me",
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "lastClassName": "gAjIeR",
        "rules": Array [
          "
  color: hsl(130, 50%, 50%);
",
        ],
      },
      "displayName": "styled.a",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "a",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Click me",
        "href": "localhost:3000",
      },
      "ref": null,
      "rendered": "Click me",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "lastClassName": "gAjIeR",
          "rules": Array [
            "
  color: hsl(130, 50%, 50%);
",
          ],
        },
        "displayName": "styled.a",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "a",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
